### PR TITLE
experiment with universal channel

### PR DESCRIPTION
**Pull request by ndelangen**
An experimental branch for a merged serverChannel into postMessageChannel

### CODE REVIEW

File: code/builders/builder-vite/src/codegen-set-addon-channel.ts [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
 
     const channel = createPostMessageChannel({ page: 'preview' });
     addons.setChannel(channel);
     window.__STORYBOOK_ADDONS_CHANNEL__ = channel;
     
     if (window.CONFIG_TYPE === 'DEVELOPMENT'){
-      const serverChannel = createWebSocketChannel({});
-      addons.setServerChannel(serverChannel);
-      window.__STORYBOOK_SERVER_CHANNEL__ = serverChannel;
+      window.__STORYBOOK_SERVER_CHANNEL__ = channel;
```
Comment (tmeasday) [inline code/builders/builder-vite/src/codegen-set-addon-channel.ts:11]: It feels weird that both channels come from `'@storybook/channel-postmessage'`. Should instead we make a new `'@storybook/channel-duplex'` or something?

**Review by tmeasday — COMMENTED** [review]
Do we make it easy to disambiguate where a channel message comes from? IIRC it was always slightly painful to figure out which iframe a message came from during composition. 

Or are you imagining that as the event names will be different for events coming from the server, you really won't need to disambiguate much?

**Review by tmeasday — COMMENTED** [review]
Do we make it easy to disambiguate where a channel message comes from? IIRC it was always slightly painful to figure out which iframe a message came from during composition. 

Or are you imagining that as the event names will be different for events coming from the server, you really won't need to disambiguate much?

**Comment by ndelangen** [comment]
> Do we make it easy to disambiguate where a channel message comes from? IIRC it was always slightly painful to figure out which iframe a message came from during composition.

This is sort of true, yes. But this is because multiple previews can emit the same message types, and then the single manager needs to reply to the originator.

This problem is not really an issue where there is a 1-to-1 relationship.

Event-types are just strings really, so any connected party can listen for/emit anything they want.

> Or are you imagining that as the event names will be different for events coming from the server, you really won't need to disambiguate much?

Any core message-types should be defined in `@storybook/core-events` IMHO.
That ensures everthing is aligned.. these message-types are just magic-strings after all...

There's always the possibility of event-types clashing.. This is why custom events should always have some unique prefix per addon. We do this for a11y-addon for example.

---

What's the use-case for knowing if an event came from the server?

File: code/builders/builder-vite/src/codegen-set-addon-channel.ts (hunk 1 above)
Comment (tmeasday) [reply code/builders/builder-vite/src/codegen-set-addon-channel.ts:11]: Appreciate that. Maybe we should rename the package then (would that be bad too?) -- or at least indicate somewhere that 'channel-postmessage' is a bit of a misnomer as this now includes 2 channels.

**Pull request merged**